presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -553,57 +554,17 @@ private PlanBuilder aggregate(PlanBuilder subPlan, QuerySpecification node)
         }
 
         // 4. Project and re-write all grouping functions
-        subPlan = handleGroupingFunctions(subPlan, analysis.getGroupingOperations(node), node, groupIdSymbol);
-
-        return subPlan;
-    }
-
-    private PlanBuilder handleGroupingFunctions(PlanBuilder subPlan, List<Expression> groupingOperations, QuerySpecification node, Optional<Symbol> groupIdSymbol)
-    {
-        if (groupingOperations.isEmpty()) {
+        if (analysis.getGroupingOperations(node).isEmpty()) {
             return subPlan;
         }
-
-        TranslationMap translations = subPlan.copyTranslations();
```

Comment:
since `appendProjections` requires expressions to be analyzed, then I think its OK to duplicate logic here.